client/home/lib/billing/components/creditcardcontainer.coffee

```diff
@@ -0,0 +1,123 @@
+_ = require 'lodash'
+{ reduxForm, formValueSelector, SubmissionError, isDirty } = require 'redux-form'
+{ connect } = require 'react-redux'
+{ createSelector } = require 'reselect'
+validator = require 'card-validator'
+whoami = require 'app/util/whoami'
+
+stripe = require 'app/redux/modules/stripe'
+bongo = require 'app/redux/modules/bongo'
+customer = require 'app/redux/modules/payment/customer'
+creditCard = require 'app/redux/modules/payment/creditcard'
+
+CreateCreditCardForm = require 'lab/CreateCreditCardForm'
+
+FORM_NAME = 'create-credit-card'
+formValue = formValueSelector(FORM_NAME)
+
+realNumber = (state) -> formValue(state, 'number')?.replace /\D/g, ''
+
+slugify = (word) -> if word then word.split(' ').join '-' else ''
+
+cardBrand = createSelector(
+  realNumber
+  (number) ->
+    brand = if number then validator.number(number).card?.type else ''
+
+    return slugify brand
+)
+
+getUserForAccount = (account, users) ->
+  return null  if not (account and users)
+
+  { nickname } = account.profile
+  [ userId ] = Object.keys(users).filter (id) ->
+    users[id].username is nickname
+
+  return if userId then users[userId] else null
+
+
+accountUser = createSelector(
+  bongo.all('JUser')
```

Comment:
what is this doing? only `JUser` you can fetch on FE for the active `JAccount`, session owner.